src/App.tsx

```diff
@@ -15,48 +15,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type SortOptions = {
+  sortType: SortType,
+  reverse: boolean,
+};
+
+export function getPreparedGoods(
+  goods: string[],
+  { sortType, reverse }: SortOptions,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortType) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case SortType.ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+
+        default:
```

Comment:
in a sorting function, you should return 0 if the two elements being compared are equal, positive value if the first is greater, and negative value if the second is greater.